src/DependencyInjection/DoctrineExtension.php

```diff
@@ -418,8 +418,8 @@ private function detectMappingType(string $directory, ContainerBuilder $containe
             }
 
             if (
-                preg_match('/^(?: \*|\/\*\*) @[a-zA-Z\\\\]*' . $quotedMappingObjectName . '\b/m', $content)
-                || preg_match('/^(?: \*|\/\*\*) @[a-zA-Z\\\\]*Embeddable\b/m', $content)
+                preg_match('/^(?: \*|\/\*\*) @\\\\?([a-zA-Z_\x80-\xff][a-zA-Z0-9_\x80-\xff]*\\\\?)*' . $quotedMappingObjectName . '\b/m', $content)
+                || preg_match('/^(?: \*|\/\*\*) @\\\\?([a-zA-Z_\x80-\xff][a-zA-Z0-9_\x80-\xff]*\\\\?)*Embeddable\b/m', $content)
```

Comment:
This regex is becoming hard to read. Please use [the x modifier](https://www.php.net/manual/en/reference.pcre.pattern.modifiers.php#:~:text=x%20(-,PCRE_EXTENDED,-)) and add comments.

Reply:
Don't worry, I'm not in a hurry :wink: